src/app/GitUI/UserControls/FileStatusList.cs

```diff
@@ -1620,56 +1576,64 @@ private void FileStatusListView_DoubleClick(object? sender, EventArgs e)
             }
         }
 
-        private void FileStatusListView_DrawSubItem(object? sender, DrawListViewSubItemEventArgs e)
+        private void FileStatusListView_DrawNode(object? sender, DrawTreeNodeEventArgs e)
```

Comment:
Incorrect marking somewhere around here
![filetree-selection](https://github.com/user-attachments/assets/90c92842-04e2-4011-8af9-15007aa800f1)

Reply:
What do you mean with "marking"?
Which different behavior do you expect?

Reply:
![image](https://github.com/user-attachments/assets/75230319-a763-476f-b99f-78f8a710d6fc)

only the beginning of the selection is shown

Reply:
The node is focused but not selected.
The selection is cleared by FormCommit.

Reply:
I find the path flattening rather confusing... If we're doing a tree, then it'd better to show trees in both lists. Otherwise, it's mentally taxing.

Reply:
Add one more configurations: tree-full with "isolated leaves". ;-)
I do not want even more configuration, keeping flat is just for those that never wants any changes.
The collapsing is a great way to quickly see which paths that are affected by a change, without using too much space.
For those that do not stick with flat, I believe they will love this quickly.

Reply:
>> only the beginning of the selection is shown

The lines and icons are drawn by the base `TreeView` control as well as some text (weird!). `FileStatusListView_DrawNode` draws the (final) text.

> The selection is cleared by FormCommit.

Focus in FormCommit is inconsistent on first click - have not understood yet.

---

> it'd better to show trees in both lists. Otherwise, it's mentally taxing.

valid point
(Astonishingly, needs more code and complete rework of `StatusSorter`. I have removed the prepared option from the toolbar button.)

Reply:
I do not like this change, the tree views are now overflowing more easily. The list is up to twice the length now (often like 50% longer). It is possible to switch view using the toolbar (biut it was better to not having to do that). 
I will survive, the treeview is a good change, approval is still there (but I have not looked at the latest changes in detail and you indicated other should review).